splash/browser_tab.py

```diff
@@ -674,32 +675,52 @@ def html(self):
         self.store_har_timing("_onHtmlRendered")
         return result
 
-    def png(self, width=None, height=None, b64=False, render_all=False,
-            scale_method=None):
-        """ Return screenshot in PNG format """
-        self.logger.log("Getting PNG: width=%s, height=%s,"
-                        " render_all=%s, scale_method=%s" %
-                        (width, height, render_all, scale_method), min_level=2)
+    @contextmanager
```

Comment:
to be more precise - because originally this method covered with `try...finally` all image conversion flow. Now it does the same, but along with code reuse allows some endpoint-specific things, like different logging message and different har timing event name